napari/movie/naparimovie.py

```diff
@@ -0,0 +1,364 @@
+        """Set the napari viewer to a given interpolated frame
+
+        Parameters
+        -------
+        frame : int
+            frame to visualize
+        """
+
+        # set view type 2D/3D and camera state
+        self.myviewer.dims.ndisplay = self.interpolated_states["ndisplay"][
+            frame
+        ]
+        self.myviewer.window.qt_viewer.view.camera.set_state(
+            self.interpolated_states["camera"][frame]
+        )
+
+        # assign interpolated visibility state
+        for j in range(len(self.myviewer.layers)):
+            self.myviewer.layers[j].visible = self.interpolated_states["vis"][
+                frame
+            ][j]
+
+        # adjust slider positions
+        for i in range(self.interpolated_states["sliders"].shape[1]):
+            self.myviewer.dims.set_point(
+                i, self.interpolated_states["sliders"][frame][i]
+            )
+
+        # update view
+        self.myviewer.window.qt_viewer.view.camera.view_changed()
+
+    def create_movie_frame(self):
+        """Create the matplotlib figure, and image object hosting all snapshots"""
+
+        newim = self.myviewer.screenshot()
+        sizes = newim.shape
+        height = float(sizes[0])
+        width = float(sizes[1])
+
+        factor = 3
+        fig = plt.figure()
+        fig.set_size_inches(factor * width / height, factor, forward=False)
+        ax = plt.Axes(fig, [0.0, 0.0, 1.0, 1.0])
+        ax.set_axis_off()
+        fig.add_axes(ax)
+
+        self.fig = fig
+        self.ax = ax
+
+        self.implot = plt.imshow(newim, animated=True)
+
+    def movie_init(self):
+        """init function for matplotlib FuncAnimation"""
+
+        newim = self.myviewer.screenshot()
+        self.implot.set_data(newim)
+        return self.implot
+
+    def update(self, frame):
+        """Update function matplotlib FuncAnimation
+
+        Parameters
+        -------
+        frame : int
+            frame to visualize
+        """
+
+        self.update_napari_state(frame)
+        newim = self.myviewer.screenshot()
+        self.implot.set_data(newim)
+        return self.implot
+
+    def make_movie(self, name="movie.mp4", resolution=600, fps=20):
+        """Create a movie based on key-frames selected in napari
+
+        Parameters
+        -------
+        name : str
+            name to use for saving the movie (can also be a path)
+        resolution: int
+            resolution in dpi to save the movie
+        fps : int
+            frames per second
+        """
+
+        # creat all states
+        self.create_steps()
+        # create movie frame
+        self.create_movie_frame()
+        # animate
+        self.anim = FuncAnimation(
+            self.fig,
+            self.update,
+            frames=np.arange(len(self.states_dict)),
+            init_func=self.movie_init,
+            blit=False,
+        )
+        plt.show()
+
+        self.anim.save(name, dpi=resolution, fps=fps)
```

Comment:
I don't think we need the matplotlib dependency here and these lines of code and instead we can just use [an imageio mp4 writer](https://imageio.readthedocs.io/en/stable/examples.html#convert-a-movie) here instead along with the `collect_images` function, similar to the approach in `make_gif`.

Maybe something like
```python
stack = self.collect_images()
name = 'name.mp4'
fps=15
writer = imageio.get_writer(name, fps=fps)

for im in stack:
    writer.append_data(im)
writer.close()
```
Is there a reason @guiwitz you didn't go with this approach and went with matplotlib instead? I havn't tried it so am not 100% sure it will work, but I think it will.

The reason we want to avoid matpotlib is that it is a rather large and complex dependency. We don't use it anywhere else in the repo right now and have no plans to, so I think we want to avoid adding it here

Reply:
To be honest I have only used matplotlib to make movies until now and didn't know imageio could be used for that. I see the point of not depending on matplotlib. I'll switch to imageio then.

Reply:
I replaced matplotlib with imageio. Making mp4 movies requires now ```pip install imageio-ffmpeg```. Not sure whether we want to include this in the dependencies or let people decide to install it on their own. Also the code is simpler now, as the same function can generate mp4 and gif, and I could get rid of several matplotlib related functions.

One thing that we might need to look into is how to adjust the resolution of the movies. I don't know how to set the resolution when capturing the napari view. The ```viewer.screenshot()``` function doesn't seem to offer options.

Reply:
> the same function can generate mp4 and gif

This is great! I hadn't realised easy to make gifs were a possibility here, and they're great for putting in README files, etc.